source/qml_v2/AdaptiveCardQmlEngine/models/AdaptiveWarning.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "pch.h"
+
+namespace AdaptiveCardQmlEngine
+{
+
+    enum class Code
+    {
+        RenderException = 1
+    };
+
+    class AdaptiveWarning
+    {
+    public:
+        AdaptiveWarning(Code code, const std::string& message);
+        Code GetStatusCode() const;
+        const std::string& GetReason() const;
+
+    private:
+        const Code m_code;
+        const std::string m_message;
+    };
+
```

Comment:
Why we need this class and also why included under models, lets connect and discuss..